mermithid/processors/misc/ComplexLineShape.py

```diff
@@ -0,0 +1,598 @@
+'''
+Fits data to complex lineshape model.
+Author: E. Machado, Y.-H. Sun, E. Novitski
+Date: 4/8/20
+'''
+
+from __future__ import absolute_import
+
+import numpy as np
+import scipy as sp
+from scipy.optimize import curve_fit
+from scipy import integrate , signal, interpolate
+import os
+import time
+import sys
+import json
+from morpho.utilities import morphologging, reader
+from morpho.processors import BaseProcessor
+from mermithid.misc import Constants
+
+logger = morphologging.getLogger(__name__)
+
+
+
+__all__ = []
+__all__.append(__name__)
+
+# Natural constants
+kr_line = 17.8260 # keV
+kr_line_width = 2.83 # eV
+e_charge = 1.60217662*10**(-19) # Coulombs , charge of electron
+m_e = 9.10938356*10**(-31) # Kilograms , mass of electron
+mass_energy_electron = 510.9989461 # keV
+
+# A lorentzian function
+def lorentzian(x_array,x0,FWHM):
+    HWHM = FWHM/2.
+    func = (1./np.pi)*HWHM/((x_array-x0)**2.+HWHM**2.)
+    return func
+
+# A gaussian function
+def gaussian(x_array,A,sigma,mu):
+    f = A*(1./(sigma*np.sqrt(2*np.pi)))*np.exp(-(((x_array-mu)/sigma)**2.)/2.)
+    return f
+
+# Converts a gaussian's FWHM to sigma
+def gaussian_FWHM_to_sigma(FWHM):
+    sigma = FWHM/(2*np.sqrt(2*np.log(2)))
+    return sigma
+
+# Converts a gaussian's sigma to FWHM
+def gaussian_sigma_to_FWHM(sigma):
+    FWHM = sigma*(2*np.sqrt(2*np.log(2)))
+    return FWHM
+
+# calculating combinatorics n choose r
+def ncr(n, r):
+    r = min(r, n-r)
+    if r == 0: return 1
+    res = 1
+    for k in range(1,r+1):
+        res = res*(n-k+1)/k
+    return res
+
+#returns array with energy loss/ oscillator strength data
+def read_oscillator_str_file(filename):
```

Comment:
in the processor'sInternalConfigure method it should already be made sure that the file read in here exists